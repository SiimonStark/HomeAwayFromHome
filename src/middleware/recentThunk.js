import {isLoading, isError, fetchRecent} from '../store/actions';

export const recentThunk = (URL) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(URL)
      if(!response.ok) {
        throw Error(response.statusText)
      }
      const data = await response.json()
      dispatch(isLoading(false))
      dispatch(fetchRecent(data))
    } catch (error) {
      dispatch(isError(error.message))
    }
  }
}