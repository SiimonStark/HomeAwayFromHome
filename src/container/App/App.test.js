import React from '../../../node_modules/react';
import ReactDOM from '../../../node_modules/react-dom';
import { App, mapDispatchToProps } from './index';
import {shallow} from 'enzyme';
import recentThunk from '../../middleware/recentThunk';

describe('App Container', () => {
  describe('App', () => {
    let wrapper;
    let mockThunk;
    beforeEach(() => {
      mockThunk = jest.fn()
      wrapper = shallow(<App recentThunk={mockThunk} />)
    })
    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    })
      // Write these tests on your own
      // You already know how!
  })
  describe('mapDispatchToProps', () => {
    // ?? Not sure how to Test Thunks in mDTP
    it.skip('calls dispatch with thunk', () => {
      let url = 'www.mock.com';
      // Setup
      const mockDispatch = jest.fn()
      const actionToDispatch = recentThunk(url)
  
      // Execution
      const mappedProps = mapDispatchToProps(mockDispatch)
  
      // Expectaion
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})