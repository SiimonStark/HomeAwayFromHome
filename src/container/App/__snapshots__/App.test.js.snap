// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Container App should match snapshot 1`] = `
<div
  className="App"
>
  <Header />
  <NavBar />
  <div
    className="display"
  >
    <Route
      component={[Function]}
      exact={true}
      to="/"
    />
    <Route
      component={[Function]}
      exact={true}
      to="/error404"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Jobboard)",
          "type": [Function],
        }
      }
      exact={true}
      to="/jobs"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Favorites)",
          "type": [Function],
        }
      }
      exact={true}
      to="/fav"
    />
  </div>
</div>
`;
